File: src/main/java/com/larryhsiao/clotho/storage/Ceres.java
package com.larryhsiao.clotho.storage;

import java.util.Map;

/**
 * Represent a key-value storage in [String].
 */
public interface Ceres {
    /**
     * Store an value with key.
     */
    void store(String key, String value);

    /**
     * Obtain value with given string.
     *
     * @return non-null value returns, not matter the key is exist or not.
     */
    String get(String key);

    /**
     * Retrieve all data in map.
     */
    Map<String, String> all();

    /**
     * Delete given key's entry.
     */
    void delete(String key);
}

File: src/main/java/com/larryhsiao/clotho/storage/Ceres.kt
package com.larryhsiao.clotho.storage

/**
 * Represent a key-value storage in [String].
 */
interface Ceres {
    /**
     * Store an value with key.
     */
    fun store(key: String, value: String)

    /**
     * Obtain value with given string.
     *
     * @return non-null value returns, not matter the key is exist or not.
     */
    fun get(key: String): String

    /**
     * Retrieve all data in map.
     */
    fun all(): Map<String, String>
}